doc/source/whatsnew/v0.25.0.rst

```diff
@@ -939,7 +939,7 @@ Performance improvements
 - Improved performance by removing the need for a garbage collect when checking for ``SettingWithCopyWarning`` (:issue:`27031`)
 - For :meth:`to_datetime` changed default value of cache parameter to ``True`` (:issue:`26043`)
 - Improved performance of :class:`DatetimeIndex` and :class:`PeriodIndex` slicing given non-unique, monotonic data (:issue:`27136`).
-
+- Improved performance of :meth:`pandas._libs.lib.maybe_convert_objects` for the case when input contains ``NaT``.
```

Comment:
Do you know what hits this from a user perspective? This is a private method which we wouldn’t mention in a what’snew

Reply:
What should I write instead?

Reply:
Depends on what would touch this from user code. I see `DataFrame.from_tuples` and maybe GroupBy ops with datetimelike objects in the result - do you see a difference when using either of those?

Reply:
best way is to run the entire asv suite (takes an hour or so)
and see what changes